azurelinuxagent/common/protocol/wire.py

```diff
@@ -867,29 +790,29 @@ def get_goal_state(self):
         return self._goal_state
 
     def get_hosting_env(self):
-        if self._hosting_env is None:
+        if self._goal_state is None:
```

Comment:
shouldn't this be `self._goal_state.hosting_env` rather than just `self._goal_state`?

Reply:
Thanks, will update it.

Reply:
The intention of this check was to ensure that update_goal_state was called before retrieving any of the properties of the goal state, wasn't it?

Reply:
Ohh sorry I missed the raise in the next line. When I introduced this code we werent raising error rather just logging a warning which is what I thought was still happening here. Thanks for clarifying!